src/libmongocrypt-compat/mongocrypt/mongocrypt.h

```diff
@@ -0,0 +1 @@
+#include "../libmongocrypt/src/mongocrypt.h"
```

Comment:
I know [MONGOCRYPT-226](https://jira.mongodb.org/browse/MONGOCRYPT-226) refers to packaging `mongocrypt-config.h` for the shared library install, but I took a look through this PR and didn't even see an instance where we include it for our bundled builds.

Do we only generate `mongocrypt-config.h` for the benefit of compiling the bundled mongocrypt sources? If we do need to include it for any reason, would it make sense to do so from this compatibility header? If we don't actually need it, then is MONGOCRYPT-226 just so we can start making use of it to report libmongocrypt's own crypto library (independent of libmongoc)?

I assume `MONGOCRYPT_VERSION` may be the only constant we reference from PHPC, and that appears to be defined in `mongocrypt.h`. Perhaps this is not an issue because we generate that by applying substitutions to `mongocrypt.h.in` -- in that case it's just another subtle difference between the libmongoc and libmongocrypt structure (since libmongoc has a separate `mongoc-version.h` file).

Reply:
> Do we only generate `mongocrypt-config.h` for the benefit of compiling the bundled mongocrypt sources? If we do need to include it for any reason, would it make sense to do so from this compatibility header? If we don't actually need it, then is MONGOCRYPT-226 just so we can start making use of it to report libmongocrypt's own crypto library (independent of libmongoc)?

We would only need it to report the crypto library, which is also why we can't rely on a forwarding header since that isn't available when compiling against system libraries. Including mongocrypt-config.h in mongocrypt.h would also solve that issue for us, since we only need to be able to check the `MONGOCRYPT_ENABLE_CRYPT` macros.

Reply:
I've updated to the latest master version and bumped the required version in config.m4. We'll wait for a release of 1.0.1 before releasing 1.7.0 👍